ComfortZoneApp/app/src/main/java/com/example/comfortzone/initial/LoginActivity.java

```diff
@@ -97,5 +116,60 @@ public void onClick(View v) {
         });
     }
 
+    private void facebookListener() {
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ParseFacebookUtils.logInWithReadPermissionsInBackground(LoginActivity.this, mPermissions, new LogInCallback() {
+                    @Override
+                    public void done(ParseUser user, ParseException e) {
+                        if (user == null) {
+                        } else if (user.isNew()) {
+                            Toast.makeText(LoginActivity.this, "Logging in, please wait", Toast.LENGTH_SHORT).show();
+                            getUserDetailFromFB();
+                        } else {
+                            goHostActivity();
+                        }
+                    }
+                });
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        ParseFacebookUtils.onActivityResult(requestCode, resultCode, data);
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    private void getUserDetailFromFB() {
+        GraphRequest request = GraphRequest.newMeRequest(AccessToken.getCurrentAccessToken(), (object, response) -> {
+            ParseUser user = ParseUser.getCurrentUser();
+            try {
+                if (object.has(KEY_NAME))
+                    user.setUsername(object.getString(KEY_NAME));
+                if (object.has(KEY_EMAIL))
+                    user.setEmail(object.getString(KEY_EMAIL));
```

Comment:
although oneline if statement without parenthesis works but it's generally discourage because it's when accidental bugs happen, so we should always put braces for if statements